apps/client/src/features/app-settings/panel/settings-panel/composite/CustomTranslationModal.tsx

```diff
@@ -0,0 +1,353 @@
+import { useMemo } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { maybeAxiosError } from '../../../../../common/api/utils';
+import Button from '../../../../../common/components/buttons/Button';
+import Input from '../../../../../common/components/input/input/Input';
+import Modal from '../../../../../common/components/modal/Modal';
+import { langEn, TranslationObject } from '../../../../../translation/languages/en';
+import { useTranslation } from '../../../../../translation/TranslationProvider';
+import * as Panel from '../../../panel-utils/PanelUtils';
+
+interface CustomTranslationModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function CustomTranslationModal({ isOpen, onClose }: CustomTranslationModalProps) {
```

Comment:
question: can we make this interface dynamic to show all the properties in the translation object?

as of now, adding a single entry would mean changes in many parts of the code base, which sounds like something we may forget to do or that can get out of sync